### ISSUE TITLE
Nextjs Only 1 Instance problem

### ISSUE THREAD

**Issue by Jervx** [kozakdenys/qr-code-styling#200]
I have this code, then I called it 2x somewhere

```
"use client"
import React, { useEffect, useRef } from "react";
import QRCodeStyling from "qr-code-styling";

import { cn } from "@/lib/utils";

type Props = {
    className?: string;
    value: string;
};

const qrCode = new QRCodeStyling({
    width: 260,
    height: 260,
    type: "svg",
    backgroundOptions : {
        color : "transparent"
    }, 
    dotsOptions: {
        color: "#4267b2",
        type: "rounded",
    },
});

const QRCode = ({ className, value }: Props) => {
    const ref = useRef<HTMLDivElement>(null);

    useEffect(()=>{
        if (!ref || !ref.current) return;
        qrCode.update({
            data: value
          });
    },[value])

    useEffect(() => {
        if (!ref || !ref.current) return;
        qrCode.append(ref.current);
    }, []);

    return (
        <div className={cn("rounded-xl overflow-clip", className)} ref={ref} />
    );
};

export default QRCode;
```

I called it 2x here

```
           <div> 
                <QRCode value={'yes'}/>
                <QRCode value={'a different one'}/>
            </div>
```

it only shows the last one
![image](https://github.com/kozakdenys/qr-code-styling/assets/48374007/fe2e47b1-ed3b-4060-8991-1ea0b285f6ac)

when I removed the last ```<QRCode value={'a different one'}/>``` this is the result

![image](https://github.com/kozakdenys/qr-code-styling/assets/48374007/20db9cbb-90a4-4ed0-88fe-216828da4971)

How can I prevent this from happening?

**Comment by Jervx** [comment]
I also tried it on the sample code which is also happens thesame

![image](https://github.com/kozakdenys/qr-code-styling/assets/48374007/67bb97fc-1617-4873-83ed-7a262cf54ad3)

**Comment by vezhdelit** [comment]
I have the same issue. No solution found yet

**Comment by korizyx** [comment]
I found a way to deal with it. Instead of svg, we'll use canvas. By some chance the library can't separate the different information for each svg, so they'll have the same value, and by switching to rendering with canvas we won't have this problem. In addition, I've also allocated to instantiate the qrcode class inside the component function, thus creating a separate instance for each qrcode, avoiding replicating the qrcode values. My code below:

**src/app/page.tsx**
```tsx
"use client";

import * as S from './styles';
import QRCodeStyling from 'qr-code-styling';

interface Props {
  value: string;
}

const QRCode = ({ value }: Props) => {
  const qrCode = new QRCodeStyling({
    width: 500,
    height: 500,
    type: 'canvas',
    backgroundOptions: {
      color: 'transparent',
    },
    dotsOptions: {
      color: '#4267b2',
      type: 'rounded',
    },
    data: value,
  });

  const ref = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (!ref || !ref.current) return;
    qrCode.update({
      data: value,
    });
  }, [value]);

  useEffect(() => {
    if (!ref || !ref.current) return;
    qrCode.append(ref.current);
  }, []);

  return <S.Qrcode ref={ref} />;
}; 

export default function Page() {
  return (
    <S.Container>
      hi
      <QRCode value={'01011010100101'} />
      other
      <QRCode value={'awdawddawdawd'} />
    </S.Container>
  );
};
```

![image](https://github.com/user-attachments/assets/af84d689-a425-4ca7-bf51-4462c7a6d407)

**Comment by kozakdenys** [comment]
Hi @Jervx @vezhdelit @korizyx the issue with 2 qr-codes on the same page was fixed in version [v1.7.1](https://github.com/kozakdenys/qr-code-styling/releases/tag/v1.7.1) and above.
Also, I have added an example of usage of this package on the [Next.js](https://github.com/kozakdenys/qr-code-styling-examples/tree/master/examples/nextjs) (server component and client)
